### PR TITLE
Added Ukrainian translation

### PR DESCRIPTION
**Pull request by DevilXD**
@Nollasko I've created a PR for you. Please verify everything's correct and in place, and let me know once you do so.

### CODE REVIEW

File: lang/Українська.json [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+      "login_to_complete": "Завершіть процедуру входу вручну, натиснувши кнопку Увійти ще раз.",
+      "no_token": "Жетон авторизації не знайдено.",
+      "closed_window": "Вікно Chrome було закрито до завершення процедури входу."
+    },
+    "error_code": "Код помилки входу: {error_code}",
+    "incorrect_login_pass": "Неправильне ім'я користувача або пароль.",
+    "incorrect_email_code": "Неправильний код електронної пошти.",
+    "incorrect_twofa_code": "Неправильний код двухфакторної аутентифікації.",
+    "email_code_required": "Потрібен код електронної пошти. Перевірте вхідний ящик.",
+    "twofa_code_required": "Потрібен жетоне двухфакторної аутентифікації."
```
Comment (Nollasko) [inline lang/Українська.json:26]: "twofa_code_required": "Потрібен жетон двухфакторної аутентифікації."

File: lang/Українська.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      "name": "Стан",
+      "idle": "Бездіяльність",
+      "exiting": "Вихід...",
+      "terminated": "Зупинено",
+      "cleanup": "Очищення каналів...",
+      "gathering": "Збір каналів...",
+      "switching": "Перемикання на канал...",
+      "fetching_inventory": "Отримання інвентарю...",
+      "fetching_campaigns": "Отримання кампаній...",
+      "adding_campaigns": "Додавання компаній до інвентарю... {counter}"
```
Comment (Nollasko) [inline lang/Українська.json:45]: "adding_campaigns": "Додавання кампаній до інвентарю... {counter}"

File: lang/Українська.json [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        "game": "Гра",
+        "viewers": "Глядачі",
+        "points": "Бали"
+      }
+    },
+    "inventory": {
+      "filter": {
+        "name": "Фільтри",
+        "show": "Показати тільки:",
+        "linked": "Ззв'язані",
```
Comment (Nollasko) [inline lang/Українська.json:110]: "linked": "Зв'язані",

File: lang/Українська.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        "tray": "Автозапуск у треї: ",
+        "tray_notifications": "Сповіщення: ",
+        "priority_only": "Тільки пріоритетні: ",
+        "proxy": "Проксі (потребує перезапуску):"
+      },
+      "game_name": "Назва гри",
+      "priority": "Пріоритет",
+      "exclude": "Виключити",
+      "reload": "Перезавантажити",
+      "reload_text": "Більшість змін потребують перезавантаження, щоб набути негайного заподії: "
```
Comment (Nollasko) [inline lang/Українська.json:147]: "reload_text": "Більшість змін потребують перезавантаження, щоб набути негайної дії: "

**Review by Nollasko — CHANGES_REQUESTED** [review]

File: lang/Українська.json [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    "chrome": {
+      "startup": "Відкриття Chrome...",
+      "login_to_complete": "Завершіть процедуру входу вручну, натиснувши кнопку Увійти ще раз.",
+      "no_token": "Жетон авторизації не знайдено.",
+      "closed_window": "Вікно Chrome було закрито до завершення процедури входу."
+    },
+    "error_code": "Код помилки входу: {error_code}",
+    "incorrect_login_pass": "Неправильне ім'я користувача або пароль.",
+    "incorrect_email_code": "Неправильний код електронної пошти.",
+    "incorrect_twofa_code": "Неправильний код двухфакторної аутентифікації.",
```
Comment (Nollasko) [inline lang/Українська.json:24]: "incorrect_twofa_code": "Неправильний код двофакторної аутентифікації.",

File: lang/Українська.json [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        "points": "Бали"
+      }
+    },
+    "inventory": {
+      "filter": {
+        "name": "Фільтри",
+        "show": "Показати тільки:",
+        "linked": "Ззв'язані",
+        "upcoming": "Наближаються",
+        "expired": "Просрочені",
```
Comment (Nollasko) [inline lang/Українська.json:112]: "expired": "Прострочені",

**Review by Nollasko — CHANGES_REQUESTED** [review]
Sorry, made few mistakes

**Comment by DevilXD** [comment]
Please review the new changes, and let me know what you think.

If you'd like to test the built application from this PR, you can download the Windows build from the CI artifacts, at the bottom of the page here: https://github.com/DevilXD/TwitchDropsMiner/actions/runs/5298922029?pr=230

File: lang/Українська.json [hunk 7]
```diff
@@ -119,7 +119,7 @@
         "not_linked": "Не зв'язано \u274c",
         "active": "Діюче \u2714",
```
Comment (Nollasko) [inline lang/Українська.json:120]: "active": "Діюча \u2714",

**Review by Nollasko — CHANGES_REQUESTED** [review]
There are strings that can't be translated due to their absence in .json file
image
image
image

File: lang/Українська.json [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+      "websocket": "Веб-сокет #{id}:",
+      "initializing": "Ініціалізація...",
+      "connected": "Підключено",
+      "disconnected": "Відключено",
+      "connecting": "З'єднання...",
+      "disconnecting": "Від'єднання...",
+      "reconnecting": "Перепідключення..."
+    },
+    "progress": {
+      "name": "Прогрес кампанії",
```
Comment (Nollasko) [inline lang/Українська.json:83]: "name": "Поступ кампанії",

File: lang/Українська.json [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+      "drop": "Дроп:",
+      "game": "Гра:",
+      "campaign": "Кампанія:",
+      "remaining": "{time} залишилося",
+      "drop_progress": "Прогрес:",
```
Comment (Nollasko) [inline lang/Українська.json:88]: "drop_progress": "Поступ:",

File: lang/Українська.json [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+      "campaign_progress": "Прогрес:"
```
Comment (Nollasko) [inline lang/Українська.json:89]: "campaign_progress": "Поступ:"

**Review by Nollasko — CHANGES_REQUESTED** [review]
I hope this is the last review, please forgive me for my inattention

**Comment by DevilXD** [comment]
> There are strings that can't be translated due to their absence in .json file

That's intentional. The About section is my own personal section, written with my own words. Translating it would be like "putting words I never said into my mouth", which is why there's no need nor way to translate those, and it should stay like that.

The "Language" option is untranslatable to allow people to find the language option and change it, even if they'd switch to a language they don't otherwise speak. Some apps translate the language change option too, and once the language switches, the user would be then unable to find the language change option again and change back otherwise.

The Twitch code message that's printed into the chat is meant to be translated at some point in the future - I haven't gotten around to making a translation key for it yet. For now, it can't be translated.

***

I've pushed out more changes as requested. Is it good now?

**Comment by Nollasko** [comment]
@DevilXD found one more untranslatable string.
image

Everything else seems right, u can close this.

**Comment by DevilXD** [comment]
> found one more untranslatable string.

It's from an unfinished, in-progress feature that I haven't ended implementing properly yet. Again, translation to-be-added later. The dev version is not stable for a reason, nothing is guaranteed to work as it should.

> Everything else seems right

Merging then. Thank you for your contribution 🙂

**Pull request merged**